packages/server/graphql/public/mutations/updateTemplateCategory.ts

```diff
@@ -0,0 +1,38 @@
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getKysely from '../../../postgres/getKysely'
+import publish from '../../../utils/publish'
+import {MutationResolvers} from '../resolverTypes'
+
+const updateTemplateCategory: MutationResolvers['updateTemplateCategory'] = async (
+  _source,
+  {templateId, mainCategory},
+  {dataLoader, socketId: mutatorId}
+) => {
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+  const pg = getKysely()
+
+  // VALIDATION
+  if (mainCategory.length < 1) return {error: {message: 'Category name cannot be blank'}}
+  if (mainCategory.length > 100) return {error: {message: 'Category name is too long'}}
+
+  // I'd like to relax this later, hence the separate validation from string length
+  const stockCategories = ['retrospective', 'estimation', 'standup', 'feedback', 'strategy']
+  if (!stockCategories.includes(mainCategory))
+    return {error: {message: 'Custom categories not available'}}
+
+  // RESOLUTION
+  await pg
+    .updateTable('MeetingTemplate')
+    .set({mainCategory})
+    .where('id', '=', templateId)
```

Comment:
this is a good DX tension...
`templateId` is guaranteed to be valid from `permissions.ts` because we compare `viewer.tms` to `template.teamId`. I like separating permissions from resolution, but I agree it's not intuitive that `templateId` has already been validated. I don't know a great way to communicate that! I added a comment for now